tests/cross/src/test/scala/tests/typedef/TypeDefinitionSuite.scala

```diff
@@ -0,0 +1,451 @@
+package tests.typedef
+
+import scala.meta.internal.jdk.CollectionConverters._
+import scala.meta.pc.OffsetParams
+
+import org.eclipse.lsp4j.Location
+import tests.pc.BasePcDefinitionSuite
+
+class TypeDefinitionSuite extends BasePcDefinitionSuite {
+
+  override def requiresJdkSources: Boolean = true
+
+  override def requiresScalaLibrarySources: Boolean = true
+
+  override def definitions(offsetParams: OffsetParams): List[Location] =
+    presentationCompiler
+      .typeDefinition(offsetParams)
+      .get()
+      .locations
+      .asScala
+      .toList
+
+  check(
+    "val",
+    """|class <<TClass>>(i: Int)
+       |
+       |object Main {
+       |  val ts@@t = new TClass(2)
+       |}""".stripMargin,
+    compat = Map(
+      "2" ->
+        """|class <<>>TClass(i: Int)
+           |
+           |object Main {
+           |  val tst = new TClass(2)
+           |}""".stripMargin
+    ),
+  )
+
+  check(
+    "for",
+    """|
+       |object Main {
+       |  for {
+       |    x <- List(1)
+       |    y <- 1.to(x)
+       |    z = y + x
+       |    if y < /*scala/Int# Int.scala*/@@x
+       |  } yield y
+       |}
+       |""".stripMargin,
+  )
+
+  check(
+    "for-flatMap",
+    """|
+       |object Main {
+       |  for {
+       |    x /*scala/Option# Option.scala*/@@<- Option(1)
+       |    y <- Option(x)
+       |  } yield y
+       |}
+       |""".stripMargin,
+  )
+
+  check(
+    "for-map",
+    """|
+       |object Main {
+       |  for {
+       |    x <- Option(1)
+       |    y /*scala/Option# Option.scala*/@@<- Option(x)
+       |  } yield y
+       |}
+       |""".stripMargin,
+  )
+
+  check(
+    "for-withFilter",
+    """|
+       |object Main {
+       |  for {
+       |    x <- Option(1)
+       |    y <- Option(x)
+       |    /*scala/Option#WithFilter# Option.scala*/@@if y > 2
+       |  } yield y
+       |}
+       |""".stripMargin,
+  )
+
+  check(
+    "constructor",
+    """
+      |class <<TClass>>(i: Int) {}
+      |
+      |object Main {
+      | def tst(m: TClass): Unit = {}
+      |
+      |  tst(new T@@Class(2))
+      |}""".stripMargin,
+    compat = Map(
+      "2" ->
+        """|class <<>>TClass(i: Int) {}
+           |
+           |object Main {
+           | def tst(m: TClass): Unit = {}
+           |
+           |  tst(new TClass(2))
+           |}
+           |""".stripMargin
+    ),
+  )
+
+  check(
+    "function",
+    """|
+       |object Main {
+       |  val increment: Int => Int = _ + 2
+       |  incre/*scala/Int# Int.scala*/@@ment(1)
+       |}
+       |""".stripMargin,
+  )
+
+  check(
+    "tuple",
+    // assert we don't go to `Tuple2.scala`
+    """|
+       |object Main {
+       |  @@(1, 2)
```

Comment:
why not?

Reply:
Hmm.... that's currently done for go to definition, so the assumption was that type definition would do the same (I copied the tests from previous attempts). Essentially, this should be handled by go to definition if ever. 

We don't handle tuples anywhere unless they are explicitly used.